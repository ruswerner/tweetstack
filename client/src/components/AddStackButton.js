import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAddIcon from 'material-ui/svg-icons/content/add';

export default class AddStackButton extends PureComponent {

  static propTypes = {
    onClick: PropTypes.func.isRequired
  };

  render() {
    return (
      <FloatingActionButton onClick={this.props.onClick}
                            secondary={true}
                            style={{
                              position: 'fixed',
                              right: 24,
                              bottom: 24,
                              zIndex: 1000
                            }}>
        <ContentAddIcon/>
      </FloatingActionButton>
    );
  }

}
