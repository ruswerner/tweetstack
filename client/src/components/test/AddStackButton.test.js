import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AddStackButton from '../AddStackButton';

describe('AddStackButton', function() {

  it('renders the button', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <AddStackButton onClick={() => null}/>
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when button is clicked', function() {
    it('should call onClick', function() {
      const onClick = sinon.spy();
      const wrapper = mount(
        <MuiThemeProvider>
          <AddStackButton onClick={onClick}/>
        </MuiThemeProvider>
      );
      wrapper.find('button').simulate('click');
      expect(onClick.callCount).toEqual(1);
    });
  });

});