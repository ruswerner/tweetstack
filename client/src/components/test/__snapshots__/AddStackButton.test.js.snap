// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStackButton renders the button 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "transparent",
      "borderRadius": "50%",
      "bottom": 24,
      "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "position": "fixed",
      "right": 24,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1000,
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ff4081",
        "border": 10,
        "borderRadius": "50%",
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 56,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": "bottom",
        "width": 56,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fill": "#ffffff",
              "height": 56,
              "lineHeight": "56px",
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;
